lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -47,7 +47,38 @@ Elixir makes a clear distinction between **documentation** and code comments. Th
 
 ## Long parameter list
 
-TODO.
+#### Problem
+
+In a functional language like Elixir, functions tend to be pure and therefore do not access or manipulate values outside their scopes. Since the input values of these functions are just their parameters, it is natural to expect that functions with a long list of parameters be created to keep them pure. However, when this list has **more than three or four parameters**, the function's interface becomes confusing and prone to errors during use.
+
+#### Example
+
+In the following example, the `loan/6` function has an unnecessarily long list of parameters, causing its interface to be confusing and potentially leading developers to introduce errors during calls to this function.
+
+```elixir
+defmodule Library do
+  # Too many parameters that can be grouped!
+  def loan(user_name, email, password, user_alias, book_title, book_ed) do
+    ...
+  end
+end
+```
+
+#### Refactoring
+
+To eliminate this anti-pattern, related parameters can be grouped using `Maps`, `Structs`, or even `Tuples`. This effectively reduces the number of function parameters, simplifying its interface. In the case of the `loan/6`, its parameters were grouped into two different `Maps`, thereby reducing its arity to `loan/2`:
+
+```elixir
+defmodule Library do
+  def loan(%{name: name, email: email, password: password, alias: alias} = user, %{title: title, ed: ed} = book) do
```

Comment:
I think we're sending some mixed signals here since this contradicts [this other point ](https://hexdocs.pm/elixir/main/code-anti-patterns.html#complex-extractions-in-clauses)?

Reply:
Although in this case we are not using guard clauses or even explaining how these extracted values will be used, I agree that we can try to improve. Do you think `def loan(%{} = user, %{} = book) do ...` would look better?

Reply:
Yes or even `def loan(user, book)` (if we do extract `%{name: name, ...} = user` in the body?
If we want to enforce specific structs, `def loan(%User{} = user, %Book{} = book)` would look good too I think, maybe we can mention both?